### ISSUE TITLE
Copying of Key Data Does not Work Correctly

### ISSUE THREAD

**Issue by sanssecours** [ElektraInitiative/libelektra#1483]
# Description of the Problem

Currently `kdb cp` does not work as expected. The following example in Markdown Shell Recorder syntax describes the problematic behaviour. 

```sh
# 🐣 Create new test data
kdb set /sw/kdb/current/plugins 🦄
#> Using name user/sw/kdb/current/plugins
#> Create a new key user/sw/kdb/current/plugins with string 🦄
kdb get /sw/kdb/current/plugins
#> 🦄

# 💾 Copy the data using a cascading namespace
kdb cp -v /sw/kdb/current/plugins /backup/plugins
#> common name: /sw/kdb/current/plugins
#> key: user/sw/kdb/current/plugins will be renamed to: /backup/pluginsgins

# 🕵🏾‍♀️ The report from `kdb cp` already looks very suspect. 
# 😢 Copying the key data did not work.
kdb get -v /backup/plugins
# RET: 1
#> got 1 keys
#> searching spec/backup/pluginsgins, found: <nothing>, options: KDB_O_CALLBACK
#>     searching proc/backup/pluginsgins, found: <nothing>, options:
#>     searching dir/backup/pluginsgins, found: <nothing>, options:
#>     searching user/backup/pluginsgins, found: <nothing>, options:
#>     searching system/backup/pluginsgins, found: <nothing>, options:
#>     searching default of spec/backup/pluginsgins, found: <nothing>, options: KDB_O_NOCASCADING
# STDERR: Did not find key
kdb get -v /backup/pluginsgins
# RET: 1
#> got 1 keys
#> searching spec/backup/pluginsgins, found: <nothing>, options: KDB_O_CALLBACK
#>     searching proc/backup/pluginsgins, found: <nothing>, options:
#>     searching dir/backup/pluginsgins, found: <nothing>, options:
#>     searching user/backup/pluginsgins, found: <nothing>, options:
#>     searching system/backup/pluginsgins, found: <nothing>, options:
#>     searching default of spec/backup/pluginsgins, found: <nothing>, options: KDB_O_NOCASCADING
# STDERR: Did not find key

# 😭 The command `kdb cp` did not create new data, although
# it exited successfully.
kdb ls user
#> user/sw/kdb/current/plugins

# 🤕 Let us try again — this time using an absolute namespace.
kdb cp /sw/kdb/current/plugins user/backup/plugins
# 🤔 Looks like this does not work either.
kdb get user/backup/plugins
# RET: 1
# STDERR: Did not find key

# 😱 Oh, there is our copy
kdb ls user
#> user/backup/pluginsgins
#> user/sw/kdb/current/plugins
kdb get user/backup/pluginsgins
#> 🦄
```

# System Information

- Elektra Version: master
- OS: macOS 10.12.4

**Comment by markus2330** [comment]
Thank you for reporting the issue!

What is the expected behavior of kdb cp/mv with cascading keys? Copying/moving all namespaces? Or only the namespace given with `-N`?

**Comment by sanssecours** [comment]
> What is the expected behavior of kdb cp/mv with cascading keys?

I would expect that `kdb` copies/moves files to the “resolved” namespace. For example, the command

```sh
kdb cp -v /sw/kdb/current/plugins /backup/plugins
```

should copy the value of `/sw/kdb/current/plugins` to `user/backup/plugins` in my opinion. It would also be okay if the command copies the value to another namespace like `spec` or `dir`. 

Basically I would expect that `kdb cp` behaves similar to the Unix utility `cp`.

> Copying/moving all namespaces?

I would expect that the command only copies/moves a single value unless we use the switch `--recursive`.

>  Or only the namespace given with `-N`?

As far as I can tell neither `kdb cp` nor `kdb mv` support the switch `-N`. However, I assume your question is, if the command should copy a value to all namespaces or only to one? If that is the case, I prefer that `kdb cp`/`kdb mv` only copies/moves data to a single namespace.

**Comment by markus2330** [comment]
I started to fix it in #1915 (by disabling cascading keys for cp&mv) but it seems like other tests are affected by this.

@sanssecours What is the best place to add your shell recorder tests? Can you add this to the PR?

**Comment by sanssecours** [comment]
> @sanssecours What is the best place to add your shell recorder tests? Can you add this to the PR?

I do not think that it makes sense to add the test provided in the problem description at all. As it currently stands, the tests only show undesired behavior.

> I started to fix it in #1915 (by disabling cascading keys for cp&mv) but it seems like other tests are affected by this.

I fixed the broken tests in PR #1917.

**Comment by markus2330** [comment]
Thank you so much!
